src/Chainweb/Miner/Coordinator.hs

```diff
@@ -131,68 +151,69 @@ newWork
     :: LogFunction
     -> ChainChoice
     -> MinerStatus
+    -> WebBlockHeaderDb
+        -- ^ this is used to lookup parent headers that are not in the cut
+        -- itself.
```

Comment:
Before it was possible to mine a new block with just the current cut and a pact service (and assuming that the pact state was in sync with the cut, no db access was needed at all in theory). 

Now, creating a new block requires access to the parent headers of the current cut. That's a major change in the chainweb algorithm. I am not completely happy with it, but I don't see a way how this could be avoided.

The reason for this is that the target computation needs the creation times of the adjacent parent headers. Before chains where autonomous and no concrete information would flow from one chain to another. The only information that was transferred was the Merkel hash. Now the creation times from neighboring chains are used.